Embrace Unity iOS Interface/EmbraceManager.swift

```diff
@@ -0,0 +1,494 @@
+                                                parent: parent,
+                                                startTime: convertDoubleToDate(ms: startTimeMs),
+                                                endTime: convertDoubleToDate(ms: endTimeMs),
+                                                attributes: attributes,
+                                                events: events,
+                                                errorCode: convertStringToErrorCode(str: errorCodeString))
+            return true
+    }
+    
+    // TODO: Reduce code duplication between handled and unhandled exceptions
+    static func logHandledException(name: String, message: String, stacktrace: String) {
+        var attributes = [
+            "exception.stacktrace": stacktrace,
+            "emb.exception_handling": "handled",
+            "emb.type": "sys.exception",
+            "exception.message": message,
+            "exception.type": name
+        ]
+        
+        Embrace.client?.log("Unity exception",
+                            severity: .error,
+                            timestamp: Date(), // Should we let users input their own exception timestamp?
+                            attributes: attributes,
+                            stackTraceBehavior: .notIncluded)
+    }
+    
+    static func logUnhandledException(name: String, message: String, stacktrace: String) {
+        var attributes = [
+            "exception.stacktrace": stacktrace,
+            "emb.exception_handling": "unhandled",
+            "emb.type": "sys.exception",
+            "exception.message": message,
+            "exception.type": name
+        ]
+        
+        Embrace.client?.log("Unity exception",
+                            severity: .error,
+                            timestamp: Date(), // Should we let users input their own exception timestamp? That would create an API mismatch among the Hosted SDKs. Let's not.
+                            attributes: attributes,
+                            stackTraceBehavior: .notIncluded)
+    }
+    
+    static func logPushNotification(title: String, body: String, subtitle: String, badge: Int, category: String) -> Bool {
+        
+        // Borrowed from Embrace Flutter SDK
+        let pushData: [AnyHashable: Any?] = [
+                        "aps": [
+                            "alert" : [
+                                "title" : title,
+                                "subtitle" : subtitle,
+                                "body" : body
+                            ],
+                            "badge" : badge as Any,
+                            "category" : category as Any
+                        ]
+                    ]
+        do {
+            try Embrace.client?.add(event: .push(userInfo: pushData as [AnyHashable: Any]))
+            return true
+        } catch let error {
+            // We should probably log this error
+        }
+        
+        return false
```

Comment:
I'd move the `return false` inside the catch block (aside of logging the error using `os_log` or internal logging)